flag_engine/segments/models.py

```diff
@@ -39,3 +39,4 @@ class SegmentModel(BaseModel):
     name: str
     rules: typing.List[SegmentRuleModel] = Field(default_factory=list)
     feature_states: typing.List[FeatureStateModel] = Field(default_factory=list)
+    meta: typing.Optional[typing.Dict[str, str]] = None
```

Comment:
I wonder if this should be at least a typed dict? Maybe one for a little further down the line when we know exactly what we need in the meta dict?